synapse/lib/ast.py

```diff
@@ -192,8 +149,15 @@ def cancel(self):
     async def iterNodePaths(self, runt):
 
         count = 0
+        subgraph = None
+
+        rules = runt.getOpt('graph')
 
-        graph = runt.getOpt('graph')
+        if rules not in (False, None):
+            if rules is True:
+                rules = {'degrees': None, 'pivots': ('-> *',)}
```

Comment:
what is the difference between degrees being None and degrees being 1?

Reply:
specifying degrees=None means that we will not limit subgraph carving based on degrees from seed nodes